src/components/Articles.jsx

```diff
@@ -13,12 +17,27 @@ export default function Articles() {
   const pageLimit = Math.ceil(totalArticles / 10)
 
   useEffect(() => {
+    if (!topic_id)
     newsApi.get(`/articles?p=${pageNumber}`)
       .then(({data}) => {
         setArticleList(data.articles)
         setTotalArticles(data.total_count)
       })
-  }, [pageNumber])
+  }, [pageNumber, topic_id])
+
+
+    useEffect(() => {
+      if (topic_id) {
+        newsApi.get(`/articles?topic=${topic_id}&p=${pageNumber}`)
+        .then(({data}) => {
+          setArticleList(data.articles)
+          setTotalArticles(data.total_count)
+        })
+      }
+    }, [topic_id, pageNumber])
```

Comment:
I don't think you need 2 useEffects here, it isn't very DRY. You can defo merge these to tidy the whole thing up